Index: kernel/classes/datatypes/ezobjectrelation/ezobjectrelationtype.php
===================================================================
--- kernel/classes/datatypes/ezobjectrelation/ezobjectrelationtype.php	(revision 16862)
+++ kernel/classes/datatypes/ezobjectrelation/ezobjectrelationtype.php	(working copy)
@@ -613,6 +613,8 @@
             if ( $relatedObjectID )
             {
                 $objectAttribute->setAttribute( 'data_int', $relatedObjectID );
+                $contentObject =& $objectAttribute->attribute( 'object' );
+                $contentObject->addContentObjectRelation( $relatedObjectID, $objectAttribute->attribute( 'version' ) );
                 return true;
             }
         }
Index: kernel/classes/datatypes/ezxmltext/ezxmltexttype.php
===================================================================
--- kernel/classes/datatypes/ezxmltext/ezxmltexttype.php	(revision 16862)
+++ kernel/classes/datatypes/ezxmltext/ezxmltexttype.php	(working copy)
@@ -598,6 +598,7 @@
         $allTags = array_merge( $links, $embeds, $embedsInline );
         $modified = false;
 
+        $contentObject =& $objectAttribute->attribute( 'object' );
         foreach( array_keys( $allTags ) as $key )
         {
             $tag =& $allTags[$key];
@@ -616,6 +617,10 @@
                 $objectID = $objectArray['id'];
                 $tag->setAttribute( 'object_id', $objectID );
                 $tag->removeAttribute( 'object_remote_id' );
+                if ( $contentObject )
+                {
+                    $contentObject->addContentObjectRelation( $objectID, $objectAttribute->attribute( 'version' ) );
+                }
                 $modified = true;
             }
             elseif ( $nodeRemoteID )
@@ -630,6 +635,11 @@
                 $nodeID = $nodeArray['node_id'];
                 $tag->setAttribute( 'node_id', $nodeID );
                 $tag->removeAttribute( 'node_remote_id' );
+                if ( $contentObject &&
+                     $node = eZContentObjectTreeNode::fetch( $nodeID ) )
+                {
+                    $contentObject->addContentObjectRelation( $node->attribute( 'contentobject_id' ), $objectAttribute->attribute( 'version' ) );
+                }
                 $modified = true;
             }
         }
Index: kernel/classes/datatypes/ezobjectrelationlist/ezobjectrelationlisttype.php
===================================================================
--- kernel/classes/datatypes/ezobjectrelationlist/ezobjectrelationlisttype.php	(revision 16862)
+++ kernel/classes/datatypes/ezobjectrelationlist/ezobjectrelationlisttype.php	(working copy)
@@ -1328,6 +1328,7 @@
 
         require_once( 'kernel/classes/ezcontentobject.php' );
         $relationItems = $relationList->elementsByName( 'relation-item' );
+        $contentObject =& $objectAttribute->attribute( 'object' );
         foreach( $relationItems as $i => $relationItem )
         {
             $relatedObjectRemoteID = $relationItem->attributeValue( 'contentobject-remote-id' );
@@ -1341,6 +1342,8 @@
                 continue;
             }
 
+            $contentObject->addContentObjectRelation( $object->attribute( 'id' ), $objectAttribute->attribute( 'version' ) );
+
             $relationItems[$i]->setAttribute( 'contentobject-id',        $object->attribute( 'id' ) );
             $relationItems[$i]->setAttribute( 'contentobject-version',   $object->attribute( 'current_version' ) );
             $relationItems[$i]->setAttribute( 'node-id',                 $object->attribute( 'main_node_id' ) );
